zigpy/application.py

```diff
@@ -863,6 +871,59 @@ def get_device_with_address(
         else:
             raise ValueError(f"Invalid address: {address!r}")
 
+    @contextlib.contextmanager
+    def callback_for_response(
+        self,
+        src: zigpy.device.Device,
+        filters: list[zigpy.typing.MatcherType],
+        callback: typing.Callable[
+            [
+                zigpy.zcl.foundation.ZCLHeader | zdo_types.ZDOHeader,
+                zigpy.zcl.foundation.CommandSchema,
+            ],
+            typing.Any,
+        ],
+    ) -> typing.Any:
+        """
+        Context manager to create a callback that is passed Zigbee responses.
+        """
+
+        listener = zigpy.util.CallbackListener(
+            device=src,
+            matchers=tuple(filters),
+            callback=callback,
+        )
+
+        self._req_listeners[src].append(listener)
+
+        try:
+            yield
```

Comment:
what is this for?

Reply:
Do you mean the purpose of the method or of the `yield` specifically?

 - It's the `@contextlib.contextmanager` code pattern that turns everything before the `yield` into `__enter__`, the object you yield into `with func() as obj`, and everything after the `yield` into `__exit__`.
 - The purpose of the callback listener is so you can temporarily attach a callback to specific packet types coming from a device, and then detach once you leave the context manager. It's used by the OTA example in the PR.

Reply:
I meant the yield itself. I need to look more at context mgrs apparently :) thx for the explanation!